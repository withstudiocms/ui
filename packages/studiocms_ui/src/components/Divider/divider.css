.sui-divider-container {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
  position: relative;
}

.sui-divider-line {
  position: absolute;
  left: 50%;
  transform: translate(-50%, 0);
  width: 100%;
  height: 1px;
  background-color: hsl(var(--border));
  z-index: 1;
  border: none;
}

.sui-divider-content {
  z-index: 2;
  color: hsl(var(--text-muted));
}
