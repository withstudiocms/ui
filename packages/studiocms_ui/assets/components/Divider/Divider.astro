---
import './divider.css';

/**
 * Props for the divider component.
 */
interface Props {
	/**
	 * The background color of the divider, used to "hide" content behind the slot. Defaults to `background-base`.
	 */
	background?: 'background-base' | 'background-step-1' | 'background-step-2' | 'background-step-3';
}

const hasDefaultSlot = Astro.slots.has('default');

const { background = 'background-base' } = Astro.props;
---
<div class="sui-divider-container">
  <hr class="sui-divider-line" />
  <div 
    class="sui-divider-content" 
    style={`background-color: hsl(var(--${background})); padding: ${hasDefaultSlot ? '.25rem .5rem;' : '0'}`}
  >
    <slot />
  </div>
</div>
