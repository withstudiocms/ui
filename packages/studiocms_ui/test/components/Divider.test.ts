import { describe, expect } from 'vitest';
import Divider from '../../src/components/Divider/Divider.astro';
import { test } from '../fixtures/vitest/AstroContainer';

describe('Divider Component', () => {
	test('renders Divider component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Divider, 'Divider');
		expect(result).toMatchSnapshot();
	});

	test.for([
		{ background: 'background-base' },
		{ background: 'background-step-1' },
		{ background: 'background-step-2' },
		{ background: 'background-step-3' },
	])('renders Divider with props "%s"', async (props, { renderComponent }) => {
		const result = await renderComponent(Divider, 'Divider', { props });
		expect(result).toMatchSnapshot();
	});

	test('renders Divider with slotted content', async ({ renderComponent }) => {
		const result = await renderComponent(Divider, 'Divider', {
			slots: { default: 'Slotted Content' },
		});
		expect(result).toMatchSnapshot();
	});
});
