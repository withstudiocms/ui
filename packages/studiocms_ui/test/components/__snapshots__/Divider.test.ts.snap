// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider Component > renders Divider component correctly 1`] = `"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-base); padding: 0">  </div> </div>"`;

exports[`Divider Component > renders Divider with props "{ background: 'background-base' }" 1`] = `"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-base); padding: 0">  </div> </div>"`;

exports[`Divider Component > renders Divider with props "{ background: 'background-step-1' }" 1`] = `"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-step-1); padding: 0">  </div> </div>"`;

exports[`Divider Component > renders Divider with props "{ background: 'background-step-2' }" 1`] = `"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-step-2); padding: 0">  </div> </div>"`;

exports[`Divider Component > renders Divider with props "{ background: 'background-step-3' }" 1`] = `"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-step-3); padding: 0">  </div> </div>"`;

exports[`Divider Component > renders Divider with slotted content 1`] = `"<div class="sui-divider-container"> <hr class="sui-divider-line"> <div class="sui-divider-content" style="background-color: var(--background-base); padding: .25rem .5rem;"> Slotted Content </div> </div>"`;
